### ISSUE TITLE
Binary cast don't works as expected

### ISSUE THREAD

**Issue by evilebottnawi** [glayzzle/php-parser#147]
Input:
```
$binary = (binary) $string;
$binary = b"binary string";
```
First parsed as `cast` node. Second as `string` node. I think it is invalid. Second should be `cast` node with option `shortForm: true` :+1:

**Comment by ichiriac** [comment]
Hi @evilebottnawi,

It's not exactly the same thing. First is a cast, converting from a type to another, the second is a property of the text. You can get more details here : https://wiki.php.net/rfc/binary_string_deprecation

The `b"text"` is a deprecated construct, meant to handle unicode support on PHP 6. 

If you encounter a string with a raw property starting with b, you probably should strip the b as it will certainly trigger an warning message. What I could do, is to expose the property `unicode: true` on the string node.

**Comment by ichiriac** [comment]
Sorry that took so long to reply. Take a look at the documentation here :
http://php.net/manual/en/language.types.type-juggling.php

> (binary) casting and b prefix forward support was added in PHP 5.2.1. Note that the (binary) cast is essential the same as (string), but it should not be relied upon.

Actually, if you cast as binary or as string it's strictly the same :

```
<?php 
$binary = (binary)$var;
$same = (string)$var;
```
I don't know for sure why this keyword has been introduced (maybe they planned to handle automatically the encoding into each string, so the cast was for disabling encoding and let strings manipulation work on binary contents) - but I'm 100% sure that everything was abandoned as the PHP tokenizer use exactly the same token on binary and string casts.

In order to have a clean code-base, you should avoid binary casting and treat them as string casting as it's more clear on how it will behave.

But as the parser may be closer to the source code, I've also introduced the `binary` cast mode + the `unicode` property on texts

**Comment by ichiriac** [comment]
This fix is now relased under 3.0.0-alpha.3
